tiledb/sm/serialization/array_schema.cc

```diff
@@ -86,69 +155,28 @@ Status filter_pipeline_to_capnp(
   for (unsigned i = 0; i < num_filters; i++) {
     const auto* filter = filter_pipeline->get_filter(i);
     auto filter_builder = filter_list_builder[i];
-    filter_builder.setType(filter_type_str(filter->type()));
-
-    switch (filter->type()) {
-      case FilterType::FILTER_BIT_WIDTH_REDUCTION: {
-        uint32_t window;
-        RETURN_NOT_OK(
-            filter->get_option(FilterOption::BIT_WIDTH_MAX_WINDOW, &window));
-        auto data = filter_builder.initData();
-        data.setUint32(window);
-        break;
-      }
-      case FilterType::FILTER_POSITIVE_DELTA: {
-        uint32_t window;
-        RETURN_NOT_OK(filter->get_option(
-            FilterOption::POSITIVE_DELTA_MAX_WINDOW, &window));
-        auto data = filter_builder.initData();
-        data.setUint32(window);
-        break;
-      }
-      case FilterType::FILTER_GZIP:
-      case FilterType::FILTER_ZSTD:
-      case FilterType::FILTER_LZ4:
-      case FilterType::FILTER_RLE:
-      case FilterType::FILTER_BZIP2:
-      case FilterType::FILTER_DOUBLE_DELTA:
-      case FilterType::FILTER_DICTIONARY: {
-        int32_t level;
-        RETURN_NOT_OK(
-            filter->get_option(FilterOption::COMPRESSION_LEVEL, &level));
-        auto data = filter_builder.initData();
-        data.setInt32(level);
-        break;
-      }
-      case FilterType::FILTER_NONE:
-      case FilterType::FILTER_BITSHUFFLE:
-      case FilterType::FILTER_BYTESHUFFLE:
-      case FilterType::FILTER_CHECKSUM_MD5:
-      case FilterType::FILTER_CHECKSUM_SHA256:
-      case FilterType::INTERNAL_FILTER_AES_256_GCM:
-      case FilterType::FILTER_SCALE_FLOAT:
-        break;
-    }
+    filter_to_capnp(filter, &filter_builder);
   }
 
   return Status::Ok();
 }
 
-static tuple<Status, optional<shared_ptr<Filter>>> filter_constructor(
-    capnp::Filter::Reader reader) {
+tuple<Status, optional<shared_ptr<Filter>>> filter_from_capnp(
```

Comment:
I like and appreciate bringing back the from_capnp naming scheme here. I have no idea if this is the best solution, but at least it's consistent will everything I've seen so far in serialization.